src/dataaggregator/aggregator/metricPoller.go

```diff
@@ -0,0 +1,122 @@
+package aggregator
+
+import (
+	"code.cloudfoundry.org/lager"
+	"dataaggregator/appmetric"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"metricscollector/metrics"
+	"metricscollector/server"
+	"net/http"
+	"strconv"
+	"time"
+)
+
+type MetricConsumer func(appMetric *appmetric.AppMetric)
+type MetricPoller struct {
+	metricCollectorUrl string
+	logger             lager.Logger
+	doneChan           chan bool
+	appChan            chan *appmetric.AppMonitor
+	metricConsumer     MetricConsumer
+	httpClient         *http.Client
+}
+
+func NewMetricPoller(metricCollectorUrl string, logger lager.Logger, appChan chan *appmetric.AppMonitor, metricConsumer MetricConsumer, httpClient *http.Client) *MetricPoller {
+	return &MetricPoller{
+		metricCollectorUrl: metricCollectorUrl,
+		logger:             logger.Session("metric-poller"),
+		appChan:            appChan,
+		doneChan:           make(chan bool),
+		metricConsumer:     metricConsumer,
+		httpClient:         httpClient,
+	}
+}
+func (m *MetricPoller) Start() {
+	go m.startMetricRetrieve()
+	m.logger.Info("metric-poller-started")
```

Comment:
This will create a metric-poller-metric-poller-started message. You have this problem all over.